### PR TITLE
Fix common::run_stateful

### PR DESCRIPTION
**Pull request by hoegaarden**
- test common::run_stateful
- share some test setup

/assign @feiskyer 
/cc @pswica

### CODE REVIEW

File: lib/testing.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/sur/bin/env bash
```
Comment (pswica) [inline lib/testing.sh]: I think it should be #!/usr/bin/env

Comment (pswica) [reply lib/testing.sh]: I also think we need the Kubernetes copyright at the top

File: lib/BUILD.bazel [hunk 2]
```diff
@@ -1,3 +1,8 @@
+sh_test(
```
Comment (pswica) [inline lib/BUILD.bazel]: This may need the copyright as well, but I'm not so sure with BUILD.bazel files...

File: lib/common_test.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# shellcheck source=../lib/common.sh
+source "$(dirname "$(readlink -ne "${BASH_SOURCE[0]}")")/common.sh"
+readonly TESTDATA="$( cd "$(dirname "$0")" && pwd )/testdata"
+
+set -e
+set -o pipefail
+
+TEST_run_stateful() {
+  # shellcheck disable=SC2034
```
Comment (pswica) [inline lib/common_test.sh]: I think we can get rid of this comment

File: lib/BUILD.bazel (hunk 2 above)
Comment (hoegaarden) [reply lib/BUILD.bazel]: I checked in k/k and most (all?) of the bazel files didn't have a copyright header.

File: lib/testing.sh (hunk 1 above)
Comment (hoegaarden) [reply lib/testing.sh]: > I also think we need the Kubernetes copyright at the top

I agree, however we have no check for that yet and many other files miss that header. That's probably another thing we need to tackle somewhen, but I don't want to do it all in this PR. I however added the header to files this PR touches / files which are related to this PR.

File: lib/common_test.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+source "$(dirname "$(readlink -ne "${BASH_SOURCE[0]}")")/testing.sh"
+
+# shellcheck source=../lib/common.sh
+source "$(dirname "$(readlink -ne "${BASH_SOURCE[0]}")")/common.sh"
+readonly TESTDATA="$( cd "$(dirname "$0")" && pwd )/testdata"
+
+set -e
+set -o pipefail
+
+TEST_run_stateful() {
```
Comment (spiffxp) [inline lib/common_test.sh]: ❤️ ❤️ ❤️

**Pull request merged**